docs/src/main/asciidoc/security-oidc-auth0-tutorial.adoc

```diff
@@ -0,0 +1,743 @@
+[id="security-oidc-auth0-tutorial"]
+= Protect Quarkus web application by using `Auth0` OpenID Connect provider
+include::_attributes.adoc[]
+:diataxis-type: tutorial
+:categories: security,web
+
+xref:security-architecture.adoc[Quarkus Security] provides comprehensive OpenId Connect (OIDC) and OAuth2 support with its `quarkus-oidc` extension, supporting both xref:security-oidc-code-flow-authentication.adoc[Authorization code flow] and xref:security-oidc-bearer-token-authentication.adoc[Bearer token] authentication mechanisms.
+
+With Quarkus, you can easily configure OIDC providers such as link:https://www.keycloak.org/documentation[Keycloak], link:https://developer.okta.com/[Okta], link:https://auth0.com/docs/[Auth0], and other xref:security-openid-connect-providers.adoc[well-known social OIDC and OAuth2 providers].
+
+Learn how to use the Quarkus OpenID Connect extension (`quarkus-oidc`) together with https://auth0.com/docs/[Auth0] OIDC provider to protect your API endpoints.
+
+== Create Auth0 application
+
+Start with creating an `Auth0` application in the `Auth0` dashboard, for example, lets create a `QuarkusAuth0` web application:
+
+image::auth0-create-application.png[Create Auth0 application]
+
+`Auth0` will create it and generate the client id and secret and show its HTTPS-based domain:
+
+image::auth0-created-application.png[Created Auth0 application]
+
+You will use these 3 properties to configure Quarkus at the next step.
+
+Next, add users to this application by selecting the `Users` option in the application's dashboard, for example:
```

Comment:
@michelle-purcell The step is indeed critical and good point, I suppose I'm hand-holding the reader :-), but I really would like to avoid any problems at all for not too experienced Auth0 users. But indeed, your suggestion is simpler and it is really obvious what to do after reading it :-). Let me apply